File: 0008. String to Integer/0008-String-to-Integer.py
class Solution:
    def myAtoi(self, s: str) -> int:
        i, ans, sign, length = 0, 0, 1, len(s)
        if not(s):
            return 0
        
        while i < length and s[i] == ' ':
            i += 1
        if i < length and s[i] == '-':
            sign = -1
            i += 1
        elif i < length and s[i] == '+':
            i += 1

        digits = set('1234567890')
        while i < length:
            if s[i] in digits:
                ans = (ans * 10) + (ord(s[i]) - 48)
                if ans >= 2147483647 and sign == 1:
                    return 2147483647
                elif ans >= 2147483648 and sign == -1:
                    return -2147483648 
                i += 1
            else:
                return ans * sign
        
        return ans * sign

File: 0008. String to Integer/0008-String-to-Integer.java
class Solution {
    public int myAtoi(String s) {
        int i = 0;
        long ans = 0, sign = 1;
        if (s.equals("")) {
            return 0;
        }

        while (i < s.length() && s.charAt(i) == ' ') {
            i += 1;
        } 
        if (i < s.length() && s.charAt(i) == '-') {
            sign = -1; i++;
        } else if (i < s.length() && s.charAt(i) == '+') {
            i++;
        }

        while (i < s.length()) {
            if (s.charAt(i) >= '0' && s.charAt(i) <= '9') {
                ans = (ans * 10) + (s.charAt(i) - '0');
                if (ans > Integer.MAX_VALUE && sign == -1) {
                    return Integer.MIN_VALUE;
                } else if (ans > Integer.MAX_VALUE && sign == 1) {
                    return Integer.MAX_VALUE;
                }
                i++;
            } else {
                return (int)(ans * sign);
            }
        }
        return (int)(ans * sign);
    }
}